src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.PerAnalyzerState.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Concurrent;
+using System.Collections.Immutable;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    /// <summary>
+    /// Stores the current partial analysis state for an analyzer.
+    /// </summary>
+    internal partial class AnalysisState
+    {
+        private class PerAnalyzerState
+        {
+            private readonly ConcurrentDictionary<CompilationEvent, AnalyzerStateData> _pendingEvents = new ConcurrentDictionary<CompilationEvent, AnalyzerStateData>();
+            private readonly ConcurrentDictionary<ISymbol, AnalyzerStateData> _pendingSymbols = new ConcurrentDictionary<ISymbol, AnalyzerStateData>();
```

Comment:
little thing. set concurrent level to 2. otherwise, concurrnet dictionary creates too many little stuff internally (due to us using thread pool) when we actually have most likely 2 concurrent thread using it concurrently.